server/libs/domain/src/facial-recognition/facial-recognition.services.ts

```diff
@@ -133,9 +139,16 @@ export class FacialRecognitionService {
       // }
 
       await this.mediaRepository.crop(asset.resizePath, output, { left, top, width: size, height: size });
-      await this.personService.update({ id: personId, thumbnailPath: output });
+      await this.personRepository.update({ id: personId, thumbnailPath: output });
     } catch (error: any) {
-      this.logger.error(`Failed to crop face for asset: ${assetId}, person: ${personId}`, error.stack);
+      this.logger.error(
+        `Failed to crop face for asset: ${assetId}, person: ${personId} - regenerating with zooming out disabled`,
+        error.stack,
+      );
+      await this.jobRepository.queue({
+        name: JobName.GENERATE_FACE_THUMBNAIL,
+        data: { ...data, zoomOut: false },
```

Comment:
It might be better to improve the algorithm to calculate the crop rather than trying bad values and catching an error.